client/blocks/app-banner/index.jsx

```diff
@@ -131,9 +132,21 @@ class AppBanner extends Component {
 			}
 		}
 
-		//TODO: update when deferred deep links are available
 		if ( this.isiOS() ) {
-			return 'itms://itunes.apple.com/us/app/wordpress/id335703880?mt=8';
+			switch ( currentSection ) {
+				case EDITOR:
+					return 'https://apps.wordpress.com/get#post';
+				case NOTES:
+					return 'https://apps.wordpress.com/get#notifications';
+				case READER:
+					if ( currentPathFragment.length === 0 ) {
+						return 'https://apps.wordpress.com/get#read';
+					}
+
+					return 'https://apps.wordpress.com/get#' + currentPathFragment;
+				case STATS:
+					return 'https://apps.wordpress.com/get#' + currentPathFragment;
+			}
```

Comment:
a more stable way of dealing with the fragment could be to encode the actual URL in its own transport and let the app disambiguate. I think at least we need some clarification in an explanatory comment _why_ we're passing the fragment and path like this.

for instance, imagine a fully-controlled URL…

```js
const routing = encodeURIComponent( JSON.stringify( {
	section: 'post',
	previousFragment: currentPathFragment,
} );
return `https://apps.wordpress.com/get?routing=${ routing }`
```

and we can make a helper for it…

```js
const urlFor = section => `https://apps.wordpress.com/get?${ encodeURIComponent( JSON.stringify( {
	section,
	previousFragment: currentPathFragment,
} ) ) }`;

case EDITOR:
	return urlFor( 'post' );

case NOTES:
	return urlFor( 'notifications' );

case READER:
	return urlFor( 'read' );
```

something like that

Reply:
I can certainly test that. The main benefit of passing the path in the way I am currently is that on the app side of things I can just pass the path directly to our routing mechanism for deep linked URLs coming from WordPress.com. It wouldn't be too much extra work to parse the necessary parts out of this type of URL though.

Reply:
I'm mainly concerned about bugs due to encoding issues in the URL

we're assuming a lot if we push out the `currentPathFragment` directly into the URL and especially if we're using a fragment as well 🤷‍♂️ 

just want to make sure that we at least recognize the places this will tend to break and determine how to or how much to mitigate them

Reply:
Thanks for raising this. I've used `encodeURIComponent` on the fragment now – please let me know if I need to do more than this.